lib/format/datetime/formatter.ex

```diff
@@ -40,22 +40,29 @@ defmodule Timex.Format.DateTime.Formatter do
   @spec lformat!(Types.valid_datetime, String.t, String.t, atom | nil) :: String.t | no_return
   def lformat!(date, format_string, locale, formatter \\ Default)
 
-  def lformat!({:error, _} = err, _format_string, _locale, _formatter),
-    do: err
+  def lformat!({:error, reason}, _format_string, _locale, _formatter),
+    do: raise TimexError, message: reason
   def lformat!(datetime, format_string, locale, :strftime),
     do: lformat!(datetime, format_string, locale, Strftime)
   def lformat!(datetime, format_string, locale, :relative),
     do: lformat!(datetime, format_string, locale, Relative)
+  def lformat!(%{__struct__: struct} = date, format_string, locale, formatter)
+    when struct in [Date, DateTime, NaiveDateTime, Time] and is_binary(format_string)
+        and is_binary(locale) and is_atom(formatter) do
+    case formatter.lformat(date, format_string, locale) do
+      {:ok, result}    -> result
+      {:error, reason} -> raise FormatError, message: reason
+    end
+  end
   def lformat!(date, format_string, locale, formatter)
-    when is_binary(format_string) and is_binary(locale) and is_atom(formatter)
-    do
-      date = case date do
-         %{__struct__: struct} when struct in [Date, DateTime, NaiveDateTime, Time] -> date
-         _other -> Timex.to_naive_datetime(date)
-      end
-      case formatter.lformat(date, format_string, locale)do
-        {:ok, result}    -> result
-        {:error, reason} -> raise FormatError, message: reason
+    when is_binary(format_string) and is_binary(locale) and is_atom(formatter) do
+    case Timex.to_naive_datetime(date) do
```

Comment:
We don't want to convert to NaiveDateTime in general here, only for things passed in which are not one of the built-in Elixir calendar types. Was there a specific reason you made this change?

Reply:
I separeted method with matching of date struct (line 49 and line 57) just for readability. I added in logic only check 
 of `Timex.to_naive_datetime(date)` result

Reply:
Sure I see now, that's fine 👍